EKF/control.cpp

```diff
@@ -756,9 +787,10 @@ void Ekf::controlHeightFusion()
 				}
 			}
 
-		} else if (_gps_data_ready && !_gps_hgt_faulty && !_in_range_aid_mode) {
+		} else if (_gps_data_ready && !_gps_hgt_faulty && _rng_hgt_faulty) {
```

Comment:
Same as in the baro check. The decision on whether we can switch to GPS should not depend on whether the ranger is bad.